### PR TITLE
OCPBUGS-21777: baremetal: populate customDeploy in advance

### PR DESCRIPTION
**Pull request by dtantsur**
Currently, we do not populate any deployment fields when creating
BareMetalHosts, relying on the Machine API to do it instead.
However, not setting customDeploy makes the BMH controller think that
we're doing an image-based deployment and configure its resources
accordingly. If MAPI is slow enough, the hosts become provisioned
before customDeploy is set, and the controller enters a loop trying and
failing to switch away from the image-based deployment.

TL;DR just set customDeploy on creation.

### CODE REVIEW

File: pkg/asset/machines/baremetal/hosts.go [hunk 1]
```diff
@@ -141,6 +141,12 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine) (*HostSet
 			// with a machine without triggering provisioning. We only
```
Comment (zaneb) [inline pkg/asset/machines/baremetal/hosts.go:141]: Should we clean up the comment about "without setting Image", since afaict that is irrelevant?

**Review by zaneb — COMMENTED** [review]
/approve

File: pkg/asset/machines/baremetal/hosts.go (hunk 1 above)
Comment (dtantsur) [reply pkg/asset/machines/baremetal/hosts.go:141]: I'll see if someone can lgtm the change right now and will either fix it right here or follow-up.
